skimage/transform/_warps.py

```diff
@@ -340,6 +342,9 @@ def _swirl_mapping(xy, center, rotation, strength, radius):
     xy[..., 0] = x0 + rho * np.cos(theta)
     xy[..., 1] = y0 + rho * np.sin(theta)
 
+    xy = np.nan_to_num(xy)
+    xy[xy <= 0] = 0
```

Comment:
@stefanv You're right, the radius check removes the NaNs. Will update in next commit.